subprojects/core/src/main/java/org/gradle/api/internal/artifacts/DependencyResolutionServices.java

```diff
@@ -15,18 +15,18 @@
  */
 package org.gradle.api.internal.artifacts;
 
-import org.gradle.api.artifacts.ConfigurationContainer;
 import org.gradle.api.artifacts.dsl.DependencyHandler;
 import org.gradle.api.artifacts.dsl.DependencyLockingHandler;
 import org.gradle.api.artifacts.dsl.RepositoryHandler;
 import org.gradle.api.attributes.AttributesSchema;
+import org.gradle.api.internal.artifacts.configurations.RoleBasedConfigurationContainerInternal;
 import org.gradle.api.internal.attributes.ImmutableAttributesFactory;
 import org.gradle.api.model.ObjectFactory;
 
 public interface DependencyResolutionServices {
     RepositoryHandler getResolveRepositoryHandler();
 
-    ConfigurationContainer getConfigurationContainer();
+    RoleBasedConfigurationContainerInternal getConfigurationContainer();
```

Comment:
I worry that this may affect injection for plugin authors. Can plugin authors already inject a `ConfigurationContainer`? If so, can they still do so after this change?

Reply:
Yes, updated test to ensure this.